src/server/protocol.ts

```diff
@@ -2158,6 +2162,11 @@ namespace ts.server.protocol {
          * and therefore may not be accurate.
          */
         isFromUncheckedFile?: true;
+        /**
+         * If true, this completion was for an auto-import of a module not yet in the program, but listed
+         * in the project package.json.
+         */
+        isPackageJsonImport?: true;

```

Comment:
This is sent so editors can report telemetry on how often this feature is actually useful. @uniqueiniquity, I heard you set something very similar up for IntelliCode completions? Does this seem like a reasonable approach, and would you be able to help me out with making the changes I need to make on the editor side?

Reply:
yep, let's plan to discuss. happy to help with consuming this in both vs and vscode.

Reply:
Maybe `packageJsonImportSource: "dependencies" | "devDependencies" | undefined`

Reply:
Hmm, we don’t trivially know whether the module came from dependencies or devDependencies. It would have to be a separate lookup. Do the editors have telemetry on configuration settings?